addons/pkg/webhooks/webhooks.go

```diff
@@ -32,14 +32,16 @@ type WebhookTLS struct {
 }
 
 const (
-	oneWeek = time.Hour * 24 * 7
-	oneDay  = time.Hour * 24
+	oneWeek               = time.Hour * 24 * 7
+	oneDay                = time.Hour * 24
+	RequireMinGracePeriod = time.Hour * 1
 )
 
 func (w *WebhookTLS) UpdateOrCreate() error {
-	if w.RotationTime > oneWeek {
-		w.Logger.Info("rotation time will be set to maximum allowed value of one Week")
-		w.RotationTime = oneWeek
+	maxAllowedRotationTime := oneWeek - RequireMinGracePeriod // ensures gracePeriod > 0
```

Comment:
The calculation of grace period is a bit confusing to me. To make it clear this is what i would propose:

1. Make the grace period or rotateBefore configurable as a flag. 
2. Default the grace period to 1 day or 1week - 1hour whichever is sane

Dont calculate the grace period anywhere . Just use the default.

Reply:
I get the logic now. Thanks for the conversation